sources/packages/backend/apps/api/src/route-controllers/overaward/_tests_/e2e/overaward.students.controller.getOverawardBalance.e2e-spec.ts

```diff
@@ -1,64 +1,100 @@
 import { HttpStatus, INestApplication } from "@nestjs/common";
 import * as request from "supertest";
-import { createFakeDisbursementOveraward } from "@sims/test-utils";
-import { Repository } from "typeorm";
+import {
+  createFakeDisbursementOveraward,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { DataSource, Repository } from "typeorm";
 import {
   DisbursementOveraward,
   DisbursementOverawardOriginType,
-  Student,
+  IdentityProviders,
 } from "@sims/sims-db";
 import {
   BEARER_AUTH_TYPE,
   createTestingAppModule,
   FakeStudentUsersTypes,
-  getStudentByFakeStudentUserType,
+  getProviderInstanceForModule,
   getStudentToken,
 } from "../../../../testHelpers";
+import { UserService } from "../../../../services";
+import { TestingModule } from "@nestjs/testing";
+import { AuthModule } from "../../../../auth/auth.module";
 
 describe("OverawardStudentsController(e2e)-getOverawardBalance", () => {
   let app: INestApplication;
-  let student: Student;
+  let appDataSource: DataSource;
+  let appModule: TestingModule;
   let disbursementOverawardRepo: Repository<DisbursementOveraward>;
 
   beforeAll(async () => {
-    const { nestApplication, dataSource } = await createTestingAppModule();
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
     app = nestApplication;
-    student = await getStudentByFakeStudentUserType(
-      FakeStudentUsersTypes.FakeStudentUserType1,
-      dataSource,
-    );
+    appDataSource = dataSource;
+    appModule = module;
     disbursementOverawardRepo = dataSource.getRepository(DisbursementOveraward);
   });
 
-  it.skip("Should return correct value for overaward balance when available.", async () => {
+  it("Should return correct value for overaward balance when available.", async () => {
     // Arrange
+    const student = await saveFakeStudent(appDataSource);
+
+    // Mock user service for auth module
+    const userService = await getProviderInstanceForModule<UserService>(
```

Comment:
The method does not need to explicitly express the generic type because it will be defined by the type of the third parameter.
![image](https://user-images.githubusercontent.com/61259237/234677623-bc954e9f-a2f0-4dbd-a3ae-bc1d0416d07a.png)